### PR TITLE
test(robot-server): test for analyses and persistence of protocol upload

### PR DESCRIPTION
**Pull request by y3rsh**
# Overview

- Test `GET` and `DELETE` `/protocols/{protocol_id}` with nonexistent `protocol_id`
- Upload a python protocol (POST /protocols) with 2 custom labware files.  Make sure the protocol is persisted on restart.

## Note
Originally we considered spawning errors and validating them for:
- protocols getting deleted from disk
- protocol directory getting deleted from disk

We validated what happens if this occurs and how to correct it but did not see the need to have a test for it.

# Risk assessment

None, test only.

### CODE REVIEW

File: robot-server/tests/integration/http_api/persistence/test_protocol_persistence.py [hunk 1]
```diff
@@ -54,3 +54,55 @@ async def test_protocols_persist(protocol: Callable[[str], IO[bytes]]) -> None:
                 == protocols_to_create
             )
             server.stop()
+
+
+async def test_protocol_and_labware_upload_persistance() -> None:
```
Comment (y3rsh) [inline robot-server/tests/integration/http_api/persistence/test_protocol_persistence.py]:
```suggestion
async def test_protocol_with_laware_upload_persistence() -> None:
```

**Comment by y3rsh** [comment]
## Please wait to review, want to have [PR](https://github.com/Opentrons/opentrons/pull/9948) in first.
- [x] merge Max's [PR](https://github.com/Opentrons/opentrons/pull/9948) into branch once on edge. 
- [x] add get nonexistent protocol id test
- [x] add delete nonexistent protocol id test
- [x] missing protocol file
- [x] extra protocol file

**Review by aptrons22 — CHANGES_REQUESTED** [review]
I just glanced over this.  I will review once completed.

File: robot-server/tests/integration/http_api/persistence/test_protocol_persistence.py [hunk 2]
```diff
@@ -54,3 +54,73 @@ async def test_protocols_persist(protocol: Callable[[str], IO[bytes]]) -> None:
...
+    async with RobotClient.make(
+        host="http://localhost", port=port, version="*"
+    ) as robot_client:
+        assert await robot_client.wait_until_dead(
+            12
+        ), "Dev Robot is running and must not be."
+        with DevServer(port=port) as server:
+            server.start()
+            assert await robot_client.wait_until_alive(
+                6
```
Comment (SyntaxColoring) [inline robot-server/tests/integration/http_api/persistence/test_protocol_persistence.py]: Can we pull the timeouts for startup and shutdown up to a shared constant somewhere (probably shared between tests), and also bump it to something like `15`? 6 is slightly too optimistic for my poor MacBook Air.

File: robot-server/tests/integration/http_api/protocols/test_persistence.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+            for p in restarted_protocols:
+                del p["analyses"]
+            uploaded_protocols.sort(key=lambda p: p["id"])
+            restarted_protocols.sort(key=lambda p: p["id"])
+            assert uploaded_protocols == restarted_protocols
+
+            server.stop()
+
+
+async def test_protocol_with_laware_upload_persistence() -> None:
```
Comment (SyntaxColoring) [inline robot-server/tests/integration/http_api/protocols/test_persistence.py]:
```suggestion
async def test_protocol_with_labware_upload_persistence() -> None:
```

**Review by SyntaxColoring — APPROVED** [review]
One typo, but looks good to merge. Thanks!

**Review by aptrons22 — APPROVED** [review]
I tested and confirmed all looks good.

**Pull request merged**